js/data/create_bucket.js

```diff
@@ -15,6 +16,10 @@ function createBucket(layer, buffers, collision, indices) {
         return;
     }
 
+    for (var k in layer.layout) {
+        layer.layout[k] = new StyleDeclaration('layout', layer.type, k, layer.layout[k]).value;
```

Comment:
`layer.layout[k] = new StyleDeclaration('layout', layer.type, k, layer.layout[k]).calculate(z)`

Pass the tile zoom as `z` into `createBucket`. Then you should be able to revert all the changes to the various individual bucket classes.